.gitignore

```diff
@@ -1 +1,2 @@
 vendor/
+composer.lock
```

Comment:
Why would you remove the lock file? Doesn't that defeat the purpose?

Reply:
in libraries it's meaningless. Even if you commit it, it's ignored when a consuming project uses the library. Makes sense if you think about it. The composer.json file defines broadly what the library works with, and a composer lock seeks to lock to specific dependency versions. Locking is great when you're building an app, but not so great when you want to share a library that can work with a range of dependency versions.

Reply:
Also there is another benefit when your testing in CI without a package lock. It allows you to test the library with multiple versions of dependencies. We're not doing that in this library yet, but we would.